### PR TITLE
Feature/ui predefined templates

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds express end points to query/get pre-defined templates.
- Modifies group-side-panel to be more generic (less verbose)
- Adds $resource endpoints to query/get predefined templates
- Modifies canvas-ctrl to load predefined templates into canvas in single click.

##### TODO:
  - After cloning/importing a template if the user decide to scrap off everything and decide to use a template what should be the behavior? Scratch everything and draw the template (which seems to be logical). If that is the case then the behavior is already there. If not we need to handle that separately.

 
predefinedtemplates

### CODE REVIEW

**Comment by elfenheart** [comment]
naming the folder adaptertemplates might be a little confusing. Since we are going away with the name adapters, should we call this folder something else?

File: cdap-ui/app/services/adapter/my-adapter-templatep-api.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+angular.module(PKG.name + '.services')
+  .factory('myAdapterTemplatesApi', function($resource) {
+    return $resource(
+      '',
+      {
+        apptype: '@apptype',
+        appname: '@appname'
+      },
+      {
+        query: {
```
Comment (elfenheart) [inline cdap-ui/app/services/adapter/my-adapter-templatep-api.js]: instead of query, should we call this list instead?

**Comment by elfenheart** [comment]
minor comment, otherwise :+1: LGTM

File: cdap-ui/app/services/adapter/my-adapter-templatep-api.js (hunk 1 above)
Comment (ajainarayanan) [inline cdap-ui/app/services/adapter/my-adapter-templatep-api.js]: They both ideally mean the same. Don't know which one is the correct usage. If query sounds not intuitive will change it though. @elfenheart thoughts?

**Comment by elfenheart** [comment]
:+1: LGTM

**Pull request merged**